Mathlib/Order/Filter/Ultrafilter.lean

```diff
@@ -414,6 +414,22 @@ protected theorem NeBot.le_pure_iff (hf : f.NeBot) : f ≤ pure a ↔ f = pure a
   ⟨Ultrafilter.unique (pure a), le_of_eq⟩
 #align filter.ne_bot.le_pure_iff Filter.NeBot.le_pure_iff
 
+protected theorem NeBot.eq_pure_iff (hf : f.NeBot) {x : α} :
+    f = pure x ↔ {x} ∈ f := by
+  rw [← hf.le_pure_iff, le_pure_iff]
+
+lemma atTop_eq_pure_of_isTop [LinearOrder α] {x : α} (hx : IsTop x) :
+    (atTop : Filter α) = pure x := by
+  have : Nonempty α := ⟨x⟩
+  have A : (atTop : Filter α).NeBot := atTop_basis.neBot_iff.2 (fun _ ↦ ⟨x, hx _⟩)
```

Comment:
Can't you use [Filter.atTop_neBot](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/AtTopBot.html#Filter.atTop_neBot) here?

Reply:
Better, thanks! I think it was just the output suggested by `exact?`, and I didn't try to simplify it further.